crates/matrix-sdk/tests/integration/send_queue.rs

```diff
@@ -471,8 +472,7 @@ async fn test_error_then_locally_reenabling() {
     // seconds).
     // It's the same transaction id that's used to signal the send error.
     let error = assert_update!(watch => error { recoverable=true, txn=txn1 });
-    let error = error.as_client_api_error().unwrap();
-    assert_eq!(error.status_code, 500);
+    assert_matches!(error, QueueWedgeError::GenericApiError { .. });
```

Comment:
This change justifies why we need to store the error itself, in the case of `GenericApiError`: otherwise here we're loosing information that it could be a client API error. Maybe `GenericApiError` could contain a (boxed or not) `matrix_sdk::Error`, as this is the most general error type we have in this crate.

Reply:
Fixed now, see here details https://github.com/matrix-org/matrix-rust-sdk/pull/4137#issuecomment-2425914825